python/src/to_rust.rs

```diff
@@ -111,3 +112,11 @@ pub fn to_rust_scalar(ob: &PyAny) -> PyResult<ScalarValue> {
         }
     })
 }
+
+pub fn to_rust_schema(ob: &PyAny) -> PyResult<Schema> {
```

Comment:
Copied from https://github.com/apache/arrow-rs/blob/master/arrow-pyarrow-integration-testing/src/lib.rs#L136

Eventually we could add an optional module to arrow-rs where we implement the PyO3 conversion traits for `arrow-rs <-> pyarrow` interoperability for easier downstream integration.